cluster-autoscaler/cloudprovider/azure/azure_container_service_pool.go

```diff
@@ -335,19 +335,19 @@ func (agentPool *ContainerServiceAgentPool) TargetSize() (int, error) {
 //SetSize contacts the underlying service and sets the size of the pool.
 //This will be called when a scale up occurs and will be called just after
 //a delete is performed from a scale down.
-func (agentPool *ContainerServiceAgentPool) SetSize(targetSize int) (err error) {
+func (agentPool *ContainerServiceAgentPool) SetSize(targetSize int, isScalingDown bool) (err error) {
 	agentPool.mutex.Lock()
 	defer agentPool.mutex.Unlock()
 
-	return agentPool.setSizeInternal(targetSize)
+	return agentPool.setSizeInternal(targetSize, isScalingDown)
 }
 
 // setSizeInternal contacts the underlying service and sets the size of the pool.
 // It should be called under lock protected.
-func (agentPool *ContainerServiceAgentPool) setSizeInternal(targetSize int) (err error) {
-	if targetSize > agentPool.MaxSize() || targetSize < agentPool.MinSize() {
-		klog.Errorf("Target size %d requested outside Max: %d, Min: %d", targetSize, agentPool.MaxSize(), agentPool.MaxSize())
-		return fmt.Errorf("target size %d requested outside Max: %d, Min: %d", targetSize, agentPool.MaxSize(), agentPool.MinSize())
+func (agentPool *ContainerServiceAgentPool) setSizeInternal(targetSize int, isScalingDown bool) (err error) {
+	if isScalingDown && targetSize < agentPool.MinSize() {
```

Comment:
also check `targetSize > agentPool.MaxSize()` for scale up

Reply:
There's no need for the checking since it has existed in function `IncreaseSize`.